src/TTGamesExplorerRebirthLib/Formats/NXGTextures.cs

```diff
@@ -67,9 +67,12 @@ private void Deserialize(byte[] buffer)
             }
 
             uint   nuTextureSetHeaderVersion  = reader.ReadUInt32BigEndian();
-            ushort nuTextureSetHeaderUnknown2 = reader.ReadUInt16BigEndian();
 
-            DateStamp = reader.ReadSized16NullTerminatedString();
+            if (nuTextureSetHeaderVersion > 0) // 0 versions don't have datestamps
```

Comment:
```suggestion
            if (nuTextureSetHeaderVersion > 0)
```
Condition is explicit enough